includes/class-give-donate-form.php

```diff
@@ -651,6 +650,45 @@ public function is_custom_price_mode() {
 
 	}
 
+	/**
+	 * Determine if custom price mode is enabled or disabled
+	 *
+	 * @since  1.8.18
+	 * @access public
+	 *
+	 * @param string|float $amount Donation Amount.
+	 *
+	 * @return bool
+	 */
+	public function is_custom_price( $amount ) {
```

Comment:
@DevinWalker I added a new helper function to detected, price_id `custom` or not on basis of amount